apps/src/aichat/views/modelCustomization/PublishNotes.tsx

```diff
@@ -70,18 +85,80 @@ const PublishNotes: React.FunctionComponent = () => {
             );
           })}
         </div>
-      )}
-      <div className={styles.footerButtonContainer}>
+      </div>
+      <div className={modelCustomizationStyles.footerButtonContainer}>
+        <Button
+          text="Save"
+          iconLeft={{iconName: 'download'}}
+          type="secondary"
+          color="black"
+          disabled={isDisabled(visibility)}
+          onClick={onSave}
+          className={modelCustomizationStyles.updateButton}
+        />
         <Button
           text="Publish"
           iconLeft={{iconName: 'upload'}}
-          disabled={isDisabled(visibility)}
-          onClick={onUpdate}
-          className={styles.updateButton}
+          disabled={
+            isDisabled(visibility) || !hasFilledOutModelCard(modelCardInfo)
+          }
+          onClick={onPublish}
+          className={modelCustomizationStyles.updateButton}
         />
       </div>
     </div>
   );
 };
 
+const getInputTag = (property: keyof ModelCardInfo) => {
+  return property === 'botName' ? 'input' : 'textarea';
+};
+
+const hasFilledOutModelCard = (modelCardInfo: ModelCardInfo) => {
```

Comment:
Would be good to move this to a selector in aichatRedux since it's a state computation that directly drives some UI